### PR TITLE
Add Hibernate support and Refactor Lesson-8 functionality

### PR DESCRIPTION
**Pull request by fabulousnesss**

### CODE REVIEW

File: src/main/java/com/fabulousnesss/myproject/dao/HibernateDAO.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.hibernate.Transaction;
+
+import java.sql.*;
+import java.util.List;
+import java.util.Optional;
+
+import static com.fabulousnesss.myproject.utils.DAOOperation.*;
+
+public abstract class HibernateDAO {
+    static void saveOrUpdate(SessionOp sOp) throws SQLException {
```
Comment (Darya-Drutsko) [inline src/main/java/com/fabulousnesss/myproject/dao/HibernateDAO.java:15]: What does mean Op?

File: src/main/java/com/fabulousnesss/myproject/dao/TicketDAO.java [hunk 2]
```diff
@@ -4,82 +4,130 @@
...
-                preparedStatements -> {
-                    preparedStatements.setInt(1, ticket.getUserId());
-                    preparedStatements.setString(2, ticket.getTicketType().name());
-                    preparedStatements.setTimestamp(3, new Timestamp(ticket.getCreationTime()));
-                    preparedStatements.executeUpdate();
-
-                    ResultSet generatedKeys = preparedStatements.getGeneratedKeys();
-                    if (generatedKeys.next()) {
-                        ticket.setId(generatedKeys.getInt(1));
+    public static void saveTicketsHibernate(List<Ticket> tickets) throws SQLException {
```
Comment (Darya-Drutsko) [inline src/main/java/com/fabulousnesss/myproject/dao/TicketDAO.java:22]: why methods are static?

File: src/main/java/com/fabulousnesss/myproject/dao/UserDAO.java [hunk 3]
```diff
@@ -2,123 +2,124 @@
...
 
-    private static final String SAVE_USER_AND_TICKET_SAVEPOINT_NAME = "SavepointUserAndTicketSave";
-    private static final String DELETE_TICKET_AND_USER_SAVEPOINT_NAME = "SavepointTicketAndUserDelete";
-
     private UserDAO() {
     }
 
-    public static Optional<User> saveUserAndTicketPostgreSQL(User user, Ticket ticket) throws SQLException {
-        return dualUpdateAndFetchPrepStatement(SAVE_USER_QUERY, SAVE_TICKET_QUERY, FETCH_USER_AND_RELATED_TICKET,
+    public static void saveUserHibernate(User user) throws SQLException {
```
Comment (Darya-Drutsko) [inline src/main/java/com/fabulousnesss/myproject/dao/UserDAO.java:21]: why methods are static?

File: src/main/java/com/fabulousnesss/myproject/dao/HibernateDAO.java (hunk 1 above)
Comment (fabulousnesss) [reply src/main/java/com/fabulousnesss/myproject/dao/HibernateDAO.java:15]: I referred to unique function behaviours as 'Operation' (=Op), indicating that some statements are involved, and used 'Session' as the object it operates on

File: src/main/java/com/fabulousnesss/myproject/dao/TicketDAO.java (hunk 2 above)
Comment (fabulousnesss) [reply src/main/java/com/fabulousnesss/myproject/dao/TicketDAO.java:22]: Initially, I didn't leverage Spring's impressive offerings, so I got by with static calls

File: src/main/java/com/fabulousnesss/myproject/dao/UserDAO.java (hunk 3 above)
Comment (fabulousnesss) [reply src/main/java/com/fabulousnesss/myproject/dao/UserDAO.java:21]: Initially, I didn't leverage Spring's impressive offerings, so I got by with static calls

**Pull request merged**